Timeline update

MarianK99 opened the pull request:
This PR updates the TimeLineManager.cs:

- manages all three timelines with one script
- added timeline if the player loses the game
- added a condition to play the timeline if the player won the game (platforms in the specific room; marked with halo-effect; player needs to step on this platform and GameState must be GameWon)
- fixes the camera rotation in the scene change from the main-menu-scene to the spacestation-scene

MarianK99 committed:
Merge branch 'develop' into timeline_camera_fix

# Conflicts:
#	Assets/Prefabs/Game/Tasks/AsteroidShooter/Rock.prefab

MarianK99 committed:
Merge remote-tracking branch 'main-repo/timeline_camera_fix' into timeline_camera_fix

MarianK99 committed:
Merge remote-tracking branch 'main-repo/timeline_camera_fix' into timeline_camera_fix

MarianK99 committed:
Merge branch 'develop' into timeline_camera_fix

MarianK99 committed:
possible solution to fix the wrong camera rotation in vr during the timelines

JPTFL committed:
synchronize camera direction in main menu and starting scene, adding endscene for losing condition

JPTFL committed:
adding endscene prefab

MarianK99 committed:
removed playing the timeline EndSceneWin after the the time is up and the integrity value > 0 -> platforms with halo in the specific room to trigger this timeline

MarianK99 committed:
Merge branch 'develop' into timeline_camera_fix

JPTFL commented on Assets/Scripts/Game/GameInformation.cs:
```diff
@@ -52,19 +52,8 @@ private void EvaluateGameState()
-                        throw new ArgumentOutOfRangeException();
-                }
+                // plays the timeline at the end of the game, when the player lost
+                if(currentGameState == GameState.GameLost) timelineManager.PlayEndSceneLose();
```
The `GameInformation` script is not intended to change the game state or execute any actions on game components, but should only observe the game state and evaluate changes to notify observers.
A more separated and cleaner way would be to add a `GameInformation`-reference to the TimelineManager-Script and listen to the OnGameStateChanged event.
E.g:
```csharp
gameInformation.OnGameStateChanged += state => {
    if(state == GameState.GameLost) PlayEndSceneLose();
};
```

JPTFL reviewed (changes requested):
Looks good to me in general, but to improve the "separation of concerns", you should move the `TimelineManager.PlayEndSceneLose` invocation from the `GameInformation` script to the `TimelineManager` script.
It would be better if you change the method documentation to the official method C# method documentation format:
```csharp
/// <summary>
/// Documentation
/// </summary>
```

Aremju reviewed (changes requested):
Also please fix merge conflicts before you consider merging.

MarianK99 committed:
moved the function execution of the end-scene (lose) from GameInformation.cs to TimelineManager.cs

MarianK99 committed:
Merge branch 'develop' into timeline_camera_fix

# Conflicts:
#	Assets/Scenes/SpaceStation.unity

MarianK99 committed:
set position after starting timeline from 0,-1,0 to 0,0,0

MarianK99 commented:
- fixed the merge conflicts resulting from the PR SoundManager update -> DONE
- moved the execution of the function to start the losing end-scene from GameInformation to TimelineManager
- changed function-comments to c#-doc

JPTFL reviewed (approved):
Good work, all requested changes implemented.
